### ISSUE TITLE
Multi-objective Optimization based on Augmecon

### ISSUE THREAD

**Issue by miladkzm** [TUMCREATE-ESTL/FLEDGE.jl#32]
Hi Sebastian,
I would like to let you know that I implemented Augmecon for the multi-objective optimization inside the solution.py module at the multi_objective_mesmo branch. The model is solved successfully in the case of non-parallel processing. 
However, when using parallel processing, there is a share_state_actor inside each function that genuinely is a dictionary that is supposed to be updated at each iteration then it should be accessible by other parallel actors to avoid double-solving some iterations. 
Therefore, it still solves the problem in the parallel mode but does not skip some problems that were supposed to be skipped. 

Basically, a new class is added to the solution.py module called Augmecon. and a new class is added to the util.py module called SharedStateActor. The model of a benchmark tri-objective problem is given in the examples/develop folder

I would highly appreciate it if you could help fix the issue at your earliest convenience.
Best regards,
Milad
